<launch>
	<!--ROS driver for joystick-->
	<node name="joy_node" pkg="joy" type="joy_node" output="screen" clear_params="true">
		<param name="dev" type="str" value="/dev/input/js0" />
	</node>
	<!-- Launches the joystick to ardrone remapping -->
	<node name="joystick_controller" pkg="ardrone_demonstrations_york" type="joystick_controller.py" output="screen" required="true">
		<!-- Configures the joystick button mapping -->
		<param name="ButtonEmergency"		value="11" />
		<param name="ButtonLand"		value="10" />
		<param name="ButtonTakeoff"		value="0" /> <!--trigger-->

		<!-- Configures the joystick axis mapping -->
		<param name="AxisRoll"			value="0" />
		<param name="AxisPitch"			value="1" />
		<param name="AxisYaw"			value="2" />
		<param name="AxisZ"			value="5" />

		<!-- Configures the joystick mapping. Joystick axe readings are divided by these numbers-->
		<param name="ScaleRoll"			value="5" />
		<param name="ScalePitch"		value="5" />
		<param name="ScaleYaw"			value="2" />
		<param name="ScaleZ"			value="1.5" />
	</node>
	
	<!-- if using a swarm drone change the ip address to the drone you're using. Otherwise remove the args="..." bit-->
	<node name="ardrone_driver" pkg="ardrone_autonomy" type="ardrone_driver" output="screen" args="-ip 192.168.1.64">
		<param name="navdata_demo" value="true" />
		<param name="looprate" value="50" />
				
		<param name="outdoor" value="0" />
		<param name="flight_without_shell" value="0" />
		<param name="enemy_colors" value="0" />
		<param name="enemy_without_shell" value="0" />
		<param name="altitude_max" value="3000" />
		<param name="altitude_min" value="300" />
	</node>		
</launch>


